simplifier/world/nodes.py

```diff
@@ -262,7 +262,8 @@ def simplify(self, keep_form: bool = False) -> Optional[WorldObject]:
         if not keep_form and (new_operation := self.factorize()):
             self.world.replace(self, new_operation)
             return new_operation
-        return self.collapse()
+        collapsed_value = self.collapse()
+        return collapsed_value if collapsed_value is not None else self
```

Comment:
This somehow violates our first assumption that `simplify` only returns something if the operand changes during the simplification (https://github.com/fkie-cad/dewolf-logic/blob/37758b53152ccf678a84c948921e0bba814686e5/simplifier/world/nodes.py#L259)
I saw that this change implied also a change in normal form conversion. I am not sure how many other places could lead to strange errors when we change it here. 
We should at least be consistent. Either simplify always returns the simplified condition or only returns the simplified condition when the operand changed. But only changing it in one place seems to be inconsitent

Reply:
I think I forgot to update the comment for the function, but in this PR I've indicated that I've made simplify always return something (otherwise we get crashes with simplifying in dewolf, as we sometimes remove operands that we still need). Have I missed out any other locations where we should return something? I've tried to cover all the functions that we should return something for simplify().

Reply:
Another option to make sure that you can simplify and always can access the simplified formula is to define it with a variable. This is what I do for the integration of logic into dewolf's restructuring. This is also the same we do in the tests. So, if I want to simplify a condition, for example `cond = (a == c) | (a != c)`, I first define a variable to it, i.e., `tmp_var = cond.world.new_variable(cond.size, tmp)` and `cond.world.define(tmp_var, condition)`. This way, the variable `tmp_var` always points to the same condition and also to the simplified condition.

Reply:
When you changed that simplify always returns the simplified formula, we should update the typing everywhere and the docstring mentioned above. Furthermore, we should check that we really always return the simplified formula, in every subroutine.

Reply:
Sorry this took a while, I was testing with some changes in dewolf. I did a force update of the branch to remove a lot of changes as I realised the RangeSimplifier already handles a lot of the problems I previously ran into.